pkg/k8s/utils/utils_test.go

```diff
@@ -45,7 +47,7 @@ func TestServiceProxyName(t *testing.T) {
 	}
 
 	// Should return only test-svc-1 which has the service-proxy-name=foo
-	optMod, _ := GetServiceAndEndpointListOptionsModifier("foo")
+	optMod, _ := GetServiceAndEndpointListOptionsModifier("foo", enableHeadlessServiceWatch)
```

Comment:
We should have a unit test for true and false?

It also feels like this test may want to be refactored to be more table driven?

Reply:
refactored these to tests to be table driven and included the missing cases, please take a look